lib/cylc/config.py

```diff
@@ -1741,11 +1739,26 @@ def generate_triggers(self, lexpression, left_nodes, right, seq,
             dependency = Dependency(expr_list, set(triggers.values()), suicide)
             self.taskdefs[right].add_dependency(dependency, seq)
 
-        # Record xtrigger labels for each task name.
-        if right not in self.xtriggers:
-            self.xtriggers[right] = xtrig_labels
-        else:
-            self.xtriggers[right] = self.xtriggers[right].union(xtrig_labels)
+        for label in xtrig_labels:
+            try:
+                xtrig = self.cfg['scheduling']['xtriggers'][label]
+            except KeyError:
+                if label == 'wall_clock':
+                    # Allow "@wall_clock" in the graph as an undeclared
+                    # zero-offset clock xtrigger.
```

Comment:
I guess this is OK as wall clock is always going to be special.

Reply:
This is just idea that you should be able to put `@wall_clock => foo` (zero offet) without bothering to declare it in the `xtriggers` section.

There's not much point in generalising this to all xtriggers with no args, because a non-clock xtrigger with no args is pretty much useless - it will only be called once (albeit repeatedly until satisfied) during the entire suite run.  (Clock xtriggers get an implicit cycle point arg, which makes them unique to the cycle point).